### PR TITLE
Test research purpose request construction [GAWB-2928]

### PR DESCRIPTION
**Pull request by aednichols**
~PR exists to trigger PR test suite.~

Not yet as comprehensive a frontend test as I had hoped.
- I was able to `(react/render (react/create-element ...))` a `ResearchPurposeSection`, call its methods, and assert on the return values - _locally, but not on Jenkins_. [Error](https://fc-jenkins.dsp-techops.broadinstitute.org/job/firecloud-ui-PR-unit-test/1301/console) (which I don't know how to debug) is `TypeError: undefined is not an object (evaluating 'type.prototype')`
- I never did figure out how to leverage React's [test utilities](https://reactjs.org/docs/test-utils.html) to run more advanced interactions. This is a problem with the mechanics of dependency imports, have gotten quite stuck.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by aednichols** [comment]
jenkins retest

File: src/cljs/test/broadfcuitest/components/research_purpose.cljs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  (testing "empty RP"
+    (is (= (library/translate-research-purpose {:ds {}}) {"NMDS" false, "NCTRL" false, "NAGR" false, "POA" false, "NCU" false, "DS" []})))
+  (testing "complicated RP"
+    (is (= (library/translate-research-purpose {:poa true,
+                                                :ds
+                                                {"http://purl.obolibrary.org/obo/DOID_0050433"
+                                                 "fatal familial insomnia",
+                                                 "http://purl.obolibrary.org/obo/DOID_4325"
+                                                 "Ebola hemorrhagic fever"},
+                                                :control true}) {"NMDS" false,
```
Comment (MattPutnam) [inline src/cljs/test/broadfcuitest/components/research_purpose.cljs]: Nit: Bump this down a line

**Review by MattPutnam — COMMENTED** [review]
👍 with a little formatting thing

**Review by davidangb — APPROVED** [review]
👍 this is great!

**Comment by aednichols** [comment]
Test failure is `AuthDomainSpec`, which should be fixed by merging dev

**Comment by aednichols** [comment]
jenkins retest

**Comment by aednichols** [comment]
jenkins retest

**Comment by aednichols** [comment]
Failed `fiab-start`

https://fc-jenkins.dsp-techops.broadinstitute.org/job/fiab-start/4389/console

**Comment by aednichols** [comment]
jenkins retest

**Comment by aednichols** [comment]
jenkins retest

**Comment by aednichols** [comment]
jenkins retest

**Comment by aednichols** [comment]
`DataSpec` failed at `with defaults and local preferences when analysis is run`

https://fc-jenkins.dsp-techops.broadinstitute.org/job/fiab-test/2116/consoleFull

**Comment by aednichols** [comment]
jenkins retest

**Pull request merged**